tests/helpers/include/data_format_inference.h

```diff
@@ -15,77 +22,230 @@ constexpr bool is_wormhole  = false;
+ *
+ * Each member represents the data format used at a specific stage:
+ * - unpack_src: unpacker input format found in L1.
+ * - unpack_dst: unpacker output format when unpacking from L1 to the register(s).
+ * - math: math format used during compute operations and storing in dest register.
+ * - pack_src: packer input format, when packing from dest register to L1.
+ * - pack_dst: packer output format, desired result format in L1.
+ */
 struct FormatConfig
 {
-    DataFormat unpack_src;
-    DataFormat unpack_dst;
-    DataFormat pack_src;
-    DataFormat pack_dst;
+    const uint32_t unpack_src;
+    const uint32_t unpack_dst;
+    const uint32_t math;
+    const uint32_t pack_src;
+    const uint32_t pack_dst;
+
+    constexpr FormatConfig(uint32_t unpack_src_, uint32_t unpack_dst_, uint32_t math_, uint32_t pack_src_, uint32_t pack_dst_) :
+        unpack_src(unpack_src_), unpack_dst(unpack_dst_), math(math_), pack_src(pack_src_), pack_dst(pack_dst_)
+    {
+    }
 };
 
 constexpr bool is_exponentB(DataFormat format)
 {
+    // Return true if format has an exponentB representation i.e 8-bit exponent
     return (format == DataFormat::Float16_b || format == DataFormat::Bfp8_b || format == DataFormat::Tf32);
 }
 
+/**
+ * Checks if the given input/output format combination is an outlier case
+ * that is unsupported by hardware and requires a workaround.
+ *
+ * This outlier case occurs when converting an 8-bit exponent format datum
+ * directly to Float16 without using an intermediate Float32 representation
+ * in the dest register.
+ *
+ * To handle this hardware limitation, the destination register stores 32-bit datums,
+ * and the packer input format is converted to Float32.
+ *
+ * @param input The input data format in L1.
+ * @param output The output data format in L1.
+ * @param is_fp32_dest_acc_en Flag indicating if 32-bit destination accumulation is enabled (dest_acc).
+ *
+ * @return true if the format combination is an unsupported hardware outlier; false otherwise.
+ */
 constexpr bool is_format_combination_outlier(DataFormat input, DataFormat output, bool is_fp32_dest_acc_en)
 {
     return (is_exponentB(input) && output == DataFormat::Float16 && !is_fp32_dest_acc_en);
 }
 
-constexpr FormatConfig get_data_formats(DataFormat input, DataFormat output, bool is_fp32_dest_acc_en)
+/**
+ * Returns the output format for the packer based on the input format in L1
+ * and whether unpacking targets the source or dest register.
+ *
+ * @tparam INPUT The input data format.
+ * @return The inferred output data format for unpacking.
+ */
+constexpr FormatConfig get_data_formats(DataFormat unpack_in, DataFormat unpack_out, DataFormat math, DataFormat pack_in, DataFormat pack_out)
 {
-    DataFormat unpack_in  = input;
-    DataFormat unpack_out = input;
-    DataFormat pack_out   = output;
-    DataFormat pack_in    = DataFormat::Invalid; // Invalid format as placeholder
+    return {
+        static_cast<uint32_t>(unpack_in),
```

Comment:
technically, the correct expression would be:
```suggestion
        static_cast<std::underlying_type_t<DataFormat>>(unpack_out),
```
and we would need to include `type_traits` header for that.

Reply:
you mean `unpack_in` right?

Reply:
yes